### PR TITLE
Query all nameservers in /etc/resolv.conf to populate ovs rules for domain names in egress policy

### PR DESCRIPTION
**Pull request by pravisankar**
- When the given domain is using DNS load balancing algorithm, depending on
which nameserver it queries, we may get different IP addr.

Fix for bug: https://bugzilla.redhat.com/show_bug.cgi?id=1575583

### CODE REVIEW

**Comment by pravisankar** [comment]
@openshift/sig-networking @knobunc @danwinship PTAL

File: pkg/network/common/dns.go [hunk 1]
```diff
@@ -189,3 +233,55 @@ func ipsEqual(oldips, newips []net.IP) bool {
 	}
 	return true
 }
+
+// getDNSResolvers extracts nameservers from the given resolver config file
+func getDNSResolvers(resolverConfigFile string) ([]string, error) {
```
Comment (smarterclayton) [inline pkg/network/common/dns.go]: miekg/dns should already have a parser for this... might be better not to write your own.

File: pkg/network/common/dns.go [hunk 2]
```diff
@@ -105,28 +117,68 @@ func (d *DNS) Update() (error, bool) {
...
+	res.ttl = minTTL
+	res.nextQueryTime = time.Now().Add(res.ttl)
+	d.dnsMap[dns] = res
+	return nil, changed
+}
+
+func (d *DNS) getIPsAndMinTTL(dns, nameserver string, minTTL time.Duration) ([]net.IP, time.Duration, error) {
+	ips := []net.IP{}
+
+	// Due to lack of any go bindings for dns resolver that actually provides TTL value, we are relying on 'dig' shell command.
```
Comment (smarterclayton) [inline pkg/network/common/dns.go]: This comment is not true, we have: https://github.com/miekg/dns

**Comment by pravisankar** [comment]
Closing this pr in favor of https://github.com/openshift/origin/pull/19805

**Pull request closed without merging**